public/app/features/dashboard-scene/utils/dashboardSessionState.ts

```diff
@@ -56,30 +61,29 @@ export function preserveDashboardSceneStateInLocalStorage(search: URLSearchParam
   if (!config.featureToggles.preserveDashboardStateWhenNavigating) {
     return;
   }
+  // console.log('preserveDashboardSceneStateInLocalStorage', search.toString());
 
   // Skipping saving state for default home dashboard
   if (!uid) {
     return;
   }
+  const queryParams: URLSearchParams = new URLSearchParams();
```

Comment:
This change is quite important. Otherwise we were saving only one value for a given param. I.e. when `?a=1&a=2` we would only save `?a=2`. Using `URLSearchParams` we will save all params, even when there are multiple ones with the same key.